frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -2053,6 +2142,77 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn emergency_throttling_works() {
+		ExtBuilder::default()
+			.onchain_fallback(false)
+			.phases(10, 10)
+			.build_and_execute(|| {
+				MultiPhase::rotate_round();
+				assert!(MultiPhase::current_phase().is_off());
+
+				// call elect while snapshot does not exist during off phase but emergency phase is
+				// throttled because not enough blocks since last election passed (phase off).
+				assert!(MultiPhase::emergency_phase_throttling());
+				assert_err!(MultiPhase::elect(), ElectionError::NothingQueued);
+				assert!(MultiPhase::current_phase().is_off());
+
+				// same as above.
+				roll_to_signed();
+				assert!(MultiPhase::emergency_phase_throttling());
+				assert_err!(MultiPhase::elect(), ElectionError::NothingQueued);
+				assert!(MultiPhase::current_phase().is_signed());
+
+				// however, if `MinBlocksBeforeEmergency` have passed since the start of the
+				// current rounf and the election fails during, EPM transitions to emergency phase.
+				let min_blocks = MinBlocksBeforeEmergency::get();
+				roll_to_signed();
```

Comment:
```suggestion
```

does not seem right. I think this test would all in all be more readable if you use `roll_to(number)` instead.